### PR TITLE
Feature/pass airflow confi as job param

### LINKED ISSUE #39002: Allow passing airflow params as job parameter in databricks job
### Description

Currently if we need to pass any parameter to spark job we need to hardcode parameters in json config. But we want to pass parameters dynamically through airflow.
This feature allows to pass airflow config paramaters as job parameters in spark job.

### Use case/motivation

Dynamically update job parameter in spark job

### Related issues

no

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by SubhamSinghal**
Fixes: #39002 

### Tests:

1. empty airflow config and empty job parameter -- empty job parameter
2. empty airflow config and some job parameter in databricks operator -- job params from operator get applied
3. non empty airflow config with empty job parameter -- airflow config gets passed as job param
4. non empty airflow config with non empty job parameter --  job params from operator get applied

### CODE REVIEW

File: airflow/providers/databricks/operators/databricks.py [hunk 1]
```diff
@@ -313,6 +313,12 @@ def execute(self, context: Context) -> int:
         if "name" not in self.json:
             raise AirflowException("Missing required parameter: name")
         job_id = self._hook.find_job_id_by_name(self.json["name"])
+        if self.json.get("parameters") is None and self.params is not None:
```
Comment (dirrao) [inline airflow/providers/databricks/operators/databricks.py]:
```suggestion
        if not self.json.get("parameters")  and self.params:
```

File: airflow/providers/databricks/operators/databricks.py [hunk 2]
```diff
@@ -313,6 +313,12 @@ def execute(self, context: Context) -> int:
...
+            job_params = self.params.items() if self.params.items() is not None else {}
```
Comment (dirrao) [inline airflow/providers/databricks/operators/databricks.py]:
```suggestion
            job_params = self.params.items() if self.params.items() else {}
```

File: airflow/providers/databricks/operators/databricks.py (hunk 1 above)
Comment (SubhamSinghal) [reply airflow/providers/databricks/operators/databricks.py]: refactored.

File: airflow/providers/databricks/operators/databricks.py (hunk 2 above)
Comment (SubhamSinghal) [reply airflow/providers/databricks/operators/databricks.py]: refactored.

**Comment by SubhamSinghal** [comment]
@dirrao Let me know if it makes more sense to add a flag "import_airflow_config" and when it is True then set airflow config as job_params in databricks jobs.

File: airflow/providers/databricks/operators/databricks.py [hunk 3]
```diff
@@ -313,6 +313,12 @@ def execute(self, context: Context) -> int:
         if "name" not in self.json:
             raise AirflowException("Missing required parameter: name")
         job_id = self._hook.find_job_id_by_name(self.json["name"])
+        if not self.json.get("parameters") and self.params:
+            job_params = self.params.items() if self.params.items() else {}
+            param_list = []
+            for k, v in job_params:
+                param_list.append({"name": k, "default": v})
+            self.json["parameters"] = param_list
```
Comment (Lee-W) [inline airflow/providers/databricks/operators/databricks.py:321]:
```suggestion
            if self.params.items() is not None:
                self.json["parameters"] = [{"name": k, "default": v} for for k, v in self.params.items()]
            else:
                self.json["parameters"] = {}
```

Comment (Lee-W) [reply airflow/providers/databricks/operators/databricks.py:321]: Where is this `self.params` from?

Comment (SubhamSinghal) [reply airflow/providers/databricks/operators/databricks.py:321]: These are airflow configs

**Pull request closed without merging**